purchases/src/defaults/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -839,6 +840,61 @@ class Purchases internal constructor(
             purchasesOrchestrator.allowSharingPlayStoreAccount = value
         }
 
+    /**
+     * The preferred UI locale override for RevenueCat UI components.
+     * This affects both API requests and UI rendering.
+     *
+     * @return The preferred UI locale override, or null if using system default
+     */
+    val preferredUILocaleOverride: String?
+        @Synchronized get() = purchasesOrchestrator.preferredUILocaleOverride
+
+    /**
+     * Override the preferred UI locale for RevenueCat UI components at runtime.
+     * This affects both API requests and UI rendering.
+     *
+     * If the locale changes, this will automatically clear the offerings cache and trigger
+     * a background refetch to get paywall templates with the correct localizations.
+     *
+     * @param localeString The locale string (e.g., "es-ES", "en-US") or null to use system default
+     * @return true if cache was cleared and refetch triggered, false if locale unchanged or cache clear rate limited
+     */
+    fun overridePreferredUILocale(localeString: String?): Boolean {
+        val previousLocale = purchasesOrchestrator.preferredUILocaleOverride
+        debugLog { "overridePreferredUILocale: changing from '$previousLocale' to '$localeString'" }
+        purchasesOrchestrator.preferredUILocaleOverride = localeString
+
+        return if (previousLocale != localeString) {
```

Comment:
I think it might be good to move this logic to the `PurchasesOrchestrator` as well.